keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1045 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary and Motivation
+
+The certificates.k8s.io API group provides a flexible, pluggable mechanism to request X.509 certificates within a Kubernetes cluster, but actually making certificates available to your workloads is left as an exercise for the reader.
+
+This KEP defines three pieces of fundamental machinery:
+* An X.509 extension for expressing pod identity
+* Node restriction support for certificate requests that use the PodIdentity
+  extension.
+* A PodCertificate projected volume source that instructs the kubelet to handle
+  provisioning a key and certificate chain on behalf of a pod.  
+Taken together, this machinery makes it feasible to securely and automatically
+deliver X.509 certificates to every pod in a cluster, without imposing an
+unreasonable burden on application developers or cluster administrators.  
+
+As a first application of this machinery, this KEP defines additional machinery that allows pods to automatically receive mTLS client certificates suitable for authenticating to kube-apiserver, iwth the same attributes provided by bound service account tokens:
+* A new TLS user-mapping capability in kube-apiserver to understand client
+  certificates that embed PodIdentity extensions.
+* A new approver/signer pair, shipped in kube-controller-manager, to issue
+  compatible client certificate to pods.
+* An enhancement to client-go's InClusterConfig that automatically loads a
+  credential bundle from a well-known filesystem path and uses is as a client
+  certificate.
+
+There is a draft implementation of the design outlined in this KEP:
+https://github.com/kubernetes/kubernetes/pull/121596
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+**Ensure that certificate issuance is accomplished without relying on bearer tokens at any step**:  One of the most significant benefits of mTLS authentication is that the credential is split into two pieces (private key and certificate), only one of which is ever transmitted to the counterparty (and thus subject to eavesdropping or theft by the counterparty).  In order to preserve this guarantee, it's important that bearer tokens are not involved *at any step*, otherwise the security of the scheme devolves to just being a bearer token.  In this KEP, as long as both kubelet and the signer implementation authenticate to kube-apiserver using their own separately-bootstrapped client certificates, then no bearer tokens are involved in certificate issuance.
```

Comment:
The API server need not be the guarantor here. It could be the cluster administrator looking at their authn / authz design and concluding that bearer tokens aren't involved in this path.

Maybe also “can't be involved“; that'd be an even stronger promise or guarantee.

Reply:
Yeah, I guess my intention here was to say that this should be possible as long as the admin sets everything up correctly.  I will clarify.

Reply:
Updated.